### ISSUE TITLE
Single Mode crashes when only a subset of variables from domain get used in combobox

### ISSUE THREAD

**Issue by matejklemen** [biolab/orange3-network#151]
##### Network version
1.4.2

##### Orange version
3.23.0.dev

##### Expected behavior
Single Mode widget doesn't crash when some variables of initial domain do not appear in combo box.

##### Actual behavior
Single Mode widget crashes. The problem seems to be that an attribute which does not have 2 unique values gets "dropped" (does not appear in `Feature:` combobox), but then `openContext(...)` is not called again with the new domain, so the whole thing crashes.  
```
ERROR:orangewidget.workflow.widgetsscheme:Error calling 'set_network' of 'Single Mode'
Traceback (most recent call last):
  File "/home/matej/orange3venv/lib/python3.6/site-packages/orangewidget/workflow/widgetsscheme.py", line 809, in process_signals_for_widget
    handler(*args)
  File "/home/matej/Documents/orange3-network/orangecontrib/network/widgets/ownxsinglemode.py", line 80, in set_network
    self.openContext(network.nodes.domain)
  File "/home/matej/orange3venv/lib/python3.6/site-packages/orangewidget/widget.py", line 913, in openContext
    self.settingsHandler.open_context(self, *a)
  File "/home/matej/Documents/orange3/Orange/widgets/settings.py", line 128, in open_context
    super().open_context(widget, domain, *self.encode_domain(domain))
  File "/home/matej/orange3venv/lib/python3.6/site-packages/orangewidget/settings.py", line 819, in open_context
    self.settings_to_widget(widget, *args)
  File "/home/matej/Documents/orange3/Orange/widgets/settings.py", line 154, in settings_to_widget
    _apply_setting(setting, instance, value)
  File "/home/matej/orange3venv/lib/python3.6/site-packages/orangewidget/settings.py", line 198, in _apply_setting
    setattr(instance, setting.name, value)
  File "/home/matej/orange3venv/lib/python3.6/site-packages/orangewidget/gui.py", line 176, in __setattr__
    callback(value)
  File "/home/matej/orange3venv/lib/python3.6/site-packages/orangewidget/gui.py", line 1784, in __call__
    self.action(*args)
  File "/home/matej/Documents/orange3/Orange/widgets/gui.py", line 719, in action
    raise ValueError("Combo box does not contain item " + repr(value))
ValueError: Combo box does not contain item DiscreteVariable(name='role', values=['admin'])
```

##### Steps to reproduce the behavior
1. Network file (`toy.net` with corresponding `toy.tsv`, both [provided here](https://github.com/biolab/orange3-network/files/3735700/toy.zip))
2. Connect with Single Mode.

##### Additional info (worksheets, data, screenshots, ...)
I think the solution is to move the line 
```python
if self.network is not None:
    self.openContext(network.nodes.domain)
```
from `set_network(...)` to the end of `_update_combos(...)` method and calling it with a new domain, composed of filtered (discrete) variables and metas.  
I would open up a PR, but I don't feel comfortable enough in my understanding of contexts to be sure that this is indeed the correct way to fix this.

**Comment by janezd** [comment]
I can no longer replicate that. Can you?

There have been some fixes in `DomainContextHandler`, which may have fixed this problem.

Calling `openContext` with the entire domain should be OK.

**Comment by matejklemen** [comment]
Yes, although not with my initial instructions, either that got fixed or it was more complex than I described.

Updated instructions (updated data here: [toy.zip](https://github.com/biolab/orange3-network/files/5899608/toy.zip)):

0. (Reset widget settings)
1. Open up toy.net in Network File.
2. Connect with Single Mode.
3. In Single Mode, select Feature: `role`, Connect: `admin`, by: `user`
4. Edit toy.tsv: change role in fourth line (ID 3) from `user` to `nan`.
5. Reload network.
Error: `ValueError: Combo box does not contain item DiscreteVariable(name='role', values=('admin',))`

Orange3: 3.28.0.dev
Orange3-network: 1.5.1

Way less common scenario than I initially described, but it seems the feature is being accessed somewhere even though it's been filtered.
